### ISSUE TITLE
Incorrect warning for array initializer indentation inside annotations when using google_checks.xml

### ISSUE THREAD

**Issue by cushon** [checkstyle/checkstyle#5951]
```java
@T.Foo({
  @T.Bar, //
})
class T {

  @interface Foo {
    Bar[] value() default {};
  }

  @interface Bar {}
}

$ cat google_checks.xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
          "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
          "http://www.puppycrawl.com/dtds/configuration_1_3.dtd">

<module name="Checker">
    <property name="charset" value="UTF-8"/>

    <module name="TreeWalker">
        <module name="Indentation">
            <property name="basicOffset" value="2"/>
            <property name="braceAdjustment" value="0"/>
            <property name="caseIndent" value="2"/>
            <property name="throwsIndent" value="4"/>
            <property name="lineWrappingIndentation" value="4"/>
            <property name="arrayInitIndent" value="2"/> <!-- !!! it is 2 for arrays-->
        </module>
    </module>
</module>
```

```
javac T.java && RUN_LOCALE="-Duser.language=en -Duser.country=US" 
java -jar checkstyle-8.10.1-all.jar -c google_checks.xml  T.java
Starting audit...
[WARN] /tmp/T.java:2: '@' has incorrect indentation level 2, expected level should be 4. [Indentation]
Audit done.
```

This formatting is allowed by the Google Java Style Guide, and should not result in a warning when using `google_checks.xml`. [§4.8.3.1](https://google.github.io/styleguide/javaguide.html#s4.8.3.1-array-initializers) allows array initializers to be formatted as "block-like constructs", and provides examples showing array initializers with a 2 space indent after the opening `{`.
full config is at https://raw.githubusercontent.com/checkstyle/checkstyle/bf2cd391c4211e0c5f4bdf60b69257ac3e17c980/src/main/resources/google_checks.xml

-------

<a href="https://www.bountysource.com/issues/59915519-incorrect-warning-for-array-initializer-indentation-inside-annotations-when-using-google_checks-xml">
<img src="https://api.bountysource.com/badge/issue?issue_id=59915519">
</a>

**Comment by romani** [comment]
this issue caused used to disable Check completely in google config.
See https://github.com/checkstyle/checkstyle/issues/6946#issuecomment-519687605

**Comment by UnicodingUnicorn** [comment]
Editing the config to change `throwsIndent` of `Indentation`'s value to `2` solves the problem. There are two ways of going about this: 
1) Having an alternate `google_checks.xml` looking for two-space indents as opposed to four.
2) Adding support for multiple-value properties so checkstyle can accept two and four space indents.
However, as mentioned in issue #3899, this issue is a temporary one caused by an inconsistency in the Google style guide. Doing 1) locks one into an "either or" scenario: there has to be only two or four space indents, or the checker will trip, even though both are technically correct. On the other hand, it probably isn't good form for a style checker to accept multiple forms for something as in 2). With both in mind, I'm going to try and implement 2) as this is a robust solution for if something like this happens again in the future.

**Comment by UnicodingUnicorn** [comment]
As it turns out, writing 2) is a massive pain due to the way the project is structured. Personally now I do not think it is worth the effort, especially for something that is a "just in case".

[Here](https://gist.github.com/UnicodingUnicorn/decd9a87a991863e7042265ca1be1be5) is the alternate config file I mention. Chuck it in `src/main/resources` and then rebuild the project and it should turn up as `google_checks_alt.xml`.

**Comment by dnsbtchr** [comment]
The alternative config works for the specific use case but it breaks others and is not compliant to the [Google style](https://google.github.io/styleguide/javaguide.html#s4.5.2-line-wrapping-indent). 

`When line-wrapping, each line after the first (each continuation line) is indented at least +4 from the original line.` 

With the new config that's not the case anymore because of `<property name="lineWrappingIndentation" value="2"/>`

**Comment by UnicodingUnicorn** [comment]
Haha, that was a far shot anyway. I'll continue to try to work on this.

**Comment by mpcabd** [comment]
To get around this issue while it's not resolved yet, you can add a comment to tell checkstyle to ignore indentation for the lines affected, like this:

```
@T.Foo({
  // CHECKSTYLE IGNORE indentation FOR NEXT 1 LINES
  @T.Bar
})
class T {

  @interface Foo {
    Bar[] value() default {};
  }

  @interface Bar {}
}
```

**Comment by Abhishek-kumar09** [comment]
@romani @rnveach 

Should `AnnotationArrayInit` follow the same rulesets as `arrayInit` does. Or it should have different rulesets.

**Comment by rnveach** [comment]
Fix was merged
